apps/orgs/utils.py

```diff
@@ -122,9 +128,10 @@ def tmp_to_builtin_org(system=0, default=0):
     else:
         raise ValueError("Must set system or default")
     ori_org = get_current_org()
-    set_current_org(org_id)
-    yield
-    if ori_org is not None:
+    try:
+        set_current_org(org_id)
+        yield
+    finally:
         set_current_org(ori_org)
 
 
```

Comment:
The provided code was cut off at October 1, 2021. Please provide the complete file to assist with checking differences and providing further insights into its usage or structure.